samples/quickperf/src/test/java/com/google/cloud/jdbc/quickperf/AppTest.java

```diff
@@ -0,0 +1,248 @@
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.exc.StreamReadException;
+import com.fasterxml.jackson.databind.DatabindException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.google.api.gax.longrunning.OperationFuture;
+
+public class AppTest {
+    private static String TEST_FILE = "src/test/resources/testfile.json";
+
+    private static GenericContainer<?> emulator;
+
+    private static String projectId = "test";
+    private static String instanceId = "test";
+    private static String databaseId = "quickperftest";
+
+    private static DatabaseAdminClient dbAdminClient;
+
+    @BeforeClass
+    public static void setup() throws InterruptedException, ExecutionException {
+        System.out.println("Starting Emulator");
+        emulator = new GenericContainer<>(
+                DockerImageName.parse("gcr.io/cloud-spanner-emulator/emulator:latest"))
+                .withExposedPorts(9010)
+                .waitingFor(Wait.forListeningPort());
+
+        emulator.setPortBindings(ImmutableList.of("9010:9010"));
+        emulator.start();
+        System.out.println("Finished starting Emulator");
+
+        List<String> ddlList = Arrays.asList(
+                "CREATE TABLE GroupMgmt ("
+                        + "group_id INT64,"
+                        + "grpname STRING(MAX),"
+                        + ") PRIMARY KEY(group_id)",
+                "CREATE TABLE Users ("
+                        + "user_id INT64,"
+                        + "name STRING(MAX),"
+                        + ") PRIMARY KEY(user_id)",
+                "CREATE TABLE membership ("
+                        + "user_id INT64,"
+                        + "group_id INT64,"
+                        + "enrolled TIMESTAMP NOT NULL OPTIONS ("
+                        + " allow_commit_timestamp = true"
+                        + "),"
+                        + ") PRIMARY KEY(user_id, group_id)");
+
+        SpannerOptions options = SpannerOptions.newBuilder()
+                .setProjectId(projectId)
+                .setEmulatorHost("localhost:" + emulator.getMappedPort(9010))
+                .setCredentials(NoCredentials.getInstance())
+                .build();
+
+        Spanner spanner = options.getService();
+        dbAdminClient = spanner.getDatabaseAdminClient();
+
+        createInstance(projectId, instanceId);
+
+        OperationFuture<Database, CreateDatabaseMetadata> op = dbAdminClient.createDatabase(
+                instanceId,
+                databaseId,
+                ddlList);
+
+        System.out.println("Creating database " + databaseId);
+        Database dbOperation = op.get();
+        System.out.println("Created database [" + databaseId + "]");
```

Comment:
You can replace all of this (and also remove the entire `createInstance` method) by using the JDBC driver for doing this as well. The `autoConfigEmulator=true` flag will automatically create the instance and the database that you specify in the JDBC connection URL, so all you need to do is something like this:

```
try (Connection connection = DriverManager.getConnection("jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s?autoConfigEmulator=true")) {
  try (Statement statement = connection.createStatement()) {
    for (String ddl : ddlList) {
      statement.addBatch(ddl);
    }
    statement.executeBatch();
  }
}
```

Reply:
fixed